pkg/types/message_envelope.go

```diff
@@ -48,6 +59,18 @@ func NewMessageEnvelope(payload []byte, ctx context.Context) MessageEnvelope {
 	return envelope
 }
 
+// NewMessageEnvelopeWithError creates a new MessageEnvelope with ErrorCode set to 1 indicating there's error
+// and the payload contains message string about the error.
+func NewMessageEnvelopeWithError(requestId string, correlationId string, errorMessage string) MessageEnvelope {
+	return MessageEnvelope{
+		CorrelationID: correlationId,
+		ApiVersion:    common.ApiVersion,
+		RequestID:     requestId,
+		ErrorCode:     1,
```

Comment:
let's discuss the ErrorCode definition in the Core WG meeting
I thought it could be the same as the http status code

Reply:
I forgot this has been defined in ADR, so let's follow the ADR statement